extensions/poetry/README.md

```diff
@@ -0,0 +1,51 @@
+# Phylum Poetry extension
+
+A [Phylum CLI](phylum-cli) extension that checks your [Poetry][poetry]
+dependencies through [Phylum][phylum] before installing them.
+
+## Installation and basic usage
+
+Clone the repository and install the extension via the Phylum CLI.
+
+```console
+git clone https://github.com/phylum-dev/cli
+phylum extension install cli/extensions/poetry
+```
+
+Prepend `phylum` to your `poetry` command invocations:
+
+```console
+phylum poetry add my-package  # This will be checked by Phylum!
+```
+
+Or set up an alias in your shell to make it transparent:
+
+```console
+alias poetry="phylum poetry"
+poetry add my-package  # This will be checked by Phylum!
+```
+
+## How it works
+
+When invoking `phylum poetry`, subcommands that would modify the
+`pyproject.toml` or `poetry.lock` files will trigger a Phylum analysis.
+
+- If the analysis is successful, the corresponding changes will be applied.
+- If the analysis is unsuccessful because some of the new dependencies don't
+  meet the required project thresholds, the command will exit with the error code
+  127.
```

Comment:
Not sure we should mention the error code explicitly here. I wouldn't want to nail myself down on that error code being 100% exactly this forever. Ultimately it shouldn't matter to users and if they need it in a script they can still check at their own risk.

Reply:
Agree -- though at some point we may want to document that there _is_ indeed a difference between the two failures, if/when we do intend to explicitly expose that to users.